market-contract/src/main.sw

```diff
@@ -857,112 +855,49 @@ fn match_order_internal(
                 true,
             )
         };
-        // emit match events
-        log_order_change_info(
+
+        // Emit events for a matched order scenario
+        emit_match_events(
             b_id,
-            OrderChangeInfo::new(
-                OrderChangeType::OrderMatched,
-                block_height(),
-                matcher,
-                tx_id(),
-                b_order
-                    .amount,
-                b_order
-                    .amount - b_amount,
-            ),
-        );
-        log(MatchOrderEvent {
-            order_id: b_id,
-            asset: get_asset_id(AssetType::Base),
-            order_matcher: matcher,
-            owner: b_order.owner,
-            counterparty: q_order.owner,
-            match_size: b_amount,
-            match_price: match_price,
-        });
-        log_order_change_info(
```

Comment:
Removed logs from `match_order_internal` and created the `emit_match_events` function.